core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/QC.java

```diff
@@ -63,6 +73,37 @@
   @JoinColumn(name = "type")
   private QcType type;
 
+  @Override
+  public int compareTo(QC o) {
+    if (type != null && !type.equals(o.getType())) {
+      return type.compareTo(o.getType());
+    }
+    return date.compareTo(o.getDate());
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)